### PR TITLE
feat: Dependency recognition

### PR DESCRIPTION
**Pull request by nvollroth**
### Summary of Changes

Currently, dependencies can only be roughly recognized. However, dependency annotations cannot yet be created.
In this regard, an improved detection of dependencies was implemented and suitably integrated into the existing system. Furthermore, suitable dependency annotations can now also be created.

### CODE REVIEW

**Comment by nvollroth** [comment]
@lars-reimann 
The code coverage of the dependency annotations is now 100%.

File: src/library_analyzer/processing/annotations/model/_annotations.py [hunk 1]
```diff
@@ -5,10 +5,12 @@
 from enum import Enum
 from typing import Any
 
+from library_analyzer.processing.api import Action, Condition
+
 ANNOTATION_SCHEMA_VERSION = 2
 
 
-class EnumReviewResult(Enum):
+class EnumReviewResult(str, Enum):
```
Comment (lars-reimann) [inline src/library_analyzer/processing/annotations/model/_annotations.py:13]: Why is this change needed?

File: src/library_analyzer/processing/annotations/_generate_dependency_annotations.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        __init__ function to be additionally examined
+
+    Returns
+    -------
+    str | None
+        Paramter id of the found paramter
+
+    """
+    if init_func is not None:
+        test_params = parameter_list + init_func.parameters
```
Comment (lars-reimann) [inline src/library_analyzer/processing/annotations/_generate_dependency_annotations.py:39]: The creation of the extra list is unnecessarily slow and this function is probably called quite often. Better have two loops, one for the parameter_list and another of init_func.parameters (if init_func is not None).

File: src/library_analyzer/processing/annotations/model/_annotations.py (hunk 1 above)
Comment (nvollroth) [reply src/library_analyzer/processing/annotations/model/_annotations.py:13]: This change came about when the problem with pytest and the enums first arose. This was an attempt to solve the problem. However, this attempt was unsuccessful. I must have forgotten to undo this change, so this is now done, as it does not add any value.

Comment (nvollroth) [reply src/library_analyzer/processing/annotations/model/_annotations.py:13]: Correction:

EnumReviewResult must additionally inherit from str, otherwise for reasons I can't explain the `test_annotation_store()` for `EnumReviewResult.NONE ` fails. I guess that the whole thing is related to the empty string.

Comment (nvollroth) [reply src/library_analyzer/processing/annotations/model/_annotations.py]: This change was originally intended to fix the problem with the enums, because at the time the change was created the enum values were not interpreted correctly without the `str `inheritance. This change solved the original problem, but caused other problems (enum values are no longer interpreted correctly when an error occurs somewhere in the code). However, this change was undone, because I noticed that I had forgotten the `.value` at the enum expressions.

**Review by lars-reimann — APPROVED** [review]
Great work, thanks!
